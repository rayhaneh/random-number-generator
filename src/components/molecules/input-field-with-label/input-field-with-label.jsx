import * as React from 'react';
import {
  InputField,
  Label
} from '../../atoms';

export const InputFieldWithLabel = (props) => {
  const {
    label,
    handleChange,
    value,
    min,
    max,
  } = props;

  return (
    <div>
      <Label text={label}/>
      <InputField
        handleChange={handleChange}
        value={value}
        min={min}
        max={max}
      />
    </div>
  );
}
