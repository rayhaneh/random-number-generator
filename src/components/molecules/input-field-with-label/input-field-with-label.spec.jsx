import * as React from 'react';
import {
  configure,
  shallow,
} from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
configure({ adapter: new Adapter() });

import { InputFieldWithLabel } from './input-field-with-label';

let props;
let comp;

describe('InputFieldWithLabel Component', () => {
  beforeEach(() => {
    props = {
      label: 'Label',
      handleChange: jest.fn(),
      placeholder: 'min',
      value: '',
    }
  });

  it('renders properly', () => {
    comp = shallow(<InputFieldWithLabel {...props} />);
    expect(comp).toMatchSnapshot();
  });
});
