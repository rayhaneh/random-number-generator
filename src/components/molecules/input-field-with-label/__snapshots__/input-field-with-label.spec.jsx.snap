// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`InputFieldWithLabel Component renders properly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <InputFieldWithLabel
    handleChange={[MockFunction]}
    label="Label"
    placeholder="min"
    value=""
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <Label
          text="Label"
        />,
        <InputField
          handleChange={[MockFunction]}
          value=""
        />,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "text": "Label",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "handleChange": [MockFunction],
          "max": undefined,
          "min": undefined,
          "value": "",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <Label
            text="Label"
          />,
          <InputField
            handleChange={[MockFunction]}
            value=""
          />,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "text": "Label",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "handleChange": [MockFunction],
            "max": undefined,
            "min": undefined,
            "value": "",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    "attachTo": undefined,
    "hydrateIn": undefined,
  },
}
`;
